src/nbl/video/CVulkanSwapchain.cpp

```diff
@@ -148,7 +148,12 @@ core::smart_refctd_ptr<CVulkanSwapchain> CVulkanSwapchain::create(core::smart_re
         return nullptr;
     }
 
-    return core::smart_refctd_ptr<CVulkanSwapchain>(new CVulkanSwapchain(std::move(device),std::move(params),imageCount,std::move(oldSwapchain),vk_swapchain,semaphores+imageCount,semaphores,semaphores+2*imageCount),core::dont_grab);
+    ISurface::SCapabilities caps = {};
+    params.surface->getSurfaceCapabilitiesForPhysicalDevice(physDev, caps);
+    // read https://registry.khronos.org/vulkan/specs/1.3-extensions/html/chap34.html#swapchain-acquire-forward-progress
+    const uint32_t maxAcquiresWithoutPresent = imageCount - caps.minImageCount + 1u;
```

Comment:
since this will be a `uint8_t` later on, you need to cap the `imageCount` to `ISwapchain::MaxImageCount` first

Reply:
there are GPUs that like to return `maxImageCount` of 2^32-1, which is why I clamp that to 255 or 256 (whatever `ISwapchain::MaxImageCount` is )

Reply:
>there are GPUs that like to return maxImageCount of 2^32-1,
oh good point, didn't know